addon/-private/class/modifier.ts

```diff
@@ -40,6 +60,10 @@ export default class ClassBasedModifier<S = DefaultSignature> {
   constructor(owner: unknown, args: ArgsFor<S>) {
     setOwner(this, owner);
     this.args = args;
+
+    deprecateForDestroyables('willDestroy', this);
+    deprecateForDestroyables('isDestroying', this);
+    deprecateForDestroyables('isDestroyed', this);
```

Comment:
I think it's very very unlikely that someone overrides the `isDestroying`/`isDestroyed` part.

Reply:
Yeah, I assume the same, but if someone *can* override those things... 😅